### PR TITLE
fread length-1 colClasses=NA_character_

### LINKED ISSUE #4237: fread colClasses=NA_character_ incorrect
Follow up to #1910. Not sure how this lasted so long.
```R
> fread("A,B\n1,3\n2,4\n", colClasses=c(NA_character_, NA_character_))
       A     B      # correct
   <int> <int>
1:     1     3
2:     2     4
> fread("A,B\n1,3\n2,4\n", colClasses=NA_character_)
        A      B    # incorrect
   <char> <char>
1:      1      3
2:      2      4
> 
```

There's a length-1 specific case at C level which doesn't deal with the NA properly when colClasses is length 1.

### PR DESCRIPTION
**Pull request by mattdowle**
Closes #4237 
Removed the branch for length 1 colClasses at C level so that logic is in one place.

### CODE REVIEW

File: src/freadR.c [hunk 1]
```diff
@@ -241,7 +241,7 @@ static void applyDrop(SEXP items, int8_t *type, int ncol, int dropSource) {
   UNPROTECT(1);
 }
 
-bool userOverride(int8_t *type, lenOff *colNames, const char *anchor, int ncol)
+bool userOverride(int8_t *type, lenOff *colNames, const char *anchor, const int ncol)
```
Comment (MichaelChirico) [inline src/freadR.c]: there's a compiler warning for declaring const incorrectly right? what about variables that _could_ be const, is that something that could be detected?

Comment (jangorecki) [reply src/freadR.c]: AFAIU more of the variables we are using should be constant as we are more commonly reading values and not writing.

Comment (MichaelChirico) [reply src/freadR.c]: yep... any tool we can use to help us increase usage?

Comment (jangorecki) [reply src/freadR.c]: I don't think it is that much important to hunt for such cases.

**Pull request merged**